pkg/network/driver_interface.go

```diff
@@ -185,6 +208,14 @@ func (di *DriverInterface) GetConnectionStats(activeBuf *ConnectionBuffer, close
 			}
 			di.moreDataErrors.Inc()
 		}
+
+		// Windows driver hashmap implementation could return this if the
+		// provided buffer is too small to contain all entries in one of
+		// the hashmap's linkedlists
+		if bytesRead == 0 && err == windows.ERROR_MORE_DATA {
+			di.resizeDriverBuffer(cap(di.readBuffer) * 2)
```

Comment:
I think in the future we should
1/ move the logic inside resizeDriverBuffer() as  readBuffer is already private to the object
2/ add a maximum increment to avoid  double the buffer on big size (like buffer = 1G I don't think 2G would be the best size, may 1G+256MB) WDYT ?